### PR TITLE
Make base-grid return an empty array by default

### PR DESCRIPTION
**Pull request by annacv**
...to avoid missing list injection Key warn 
Captura de pantalla 2024-05-30 a les 13 16 36

### CODE REVIEW

**Review by joseacabaneros — DISMISSED** [review]

File: packages/x-components/src/components/base-grid.vue [hunk 1]
```diff
@@ -114,7 +114,7 @@
        *
        * @internal
        */
-      const injectedListItems = inject<Ref<ListItem[]>>(LIST_ITEMS_KEY as string);
+      const injectedListItems = inject<Ref<ListItem[]> | []>(LIST_ITEMS_KEY as string, []);
```
Comment (joseacabaneros) [inline packages/x-components/src/components/base-grid.vue:117]: Array type inside `Ref`: `inject<Ref<ListItem[] | []>>`

Comment (annacv) [reply packages/x-components/src/components/base-grid.vue:117]: Hi, wasn't sure about it as ID warned: 
Captura de pantalla 2024-06-05 a les 18 28 07

Comment (annacv) [reply packages/x-components/src/components/base-grid.vue]: ☝️ Set ref([]) as default value made the job, but , what do we want to do with the console.warn? Keep it or shall I change tests?? 🙏 @joseacabaneros

Comment (joseacabaneros) [reply packages/x-components/src/components/base-grid.vue]: I'm checking how we implement this same case in other injections, and it is common to use `undefined` as default value for the `inject`. 
With the aim of align the code base, could you change it by `undefined`? 
It is all about alignment, yours is also perfectly valid  ✌🏾  

```TS
const injectedListItems = inject<Ref<ListItem[]> | undefined>(
  LIST_ITEMS_KEY as string,
  undefined
);
```

**Review by joseacabaneros — APPROVED** [review]

**Pull request merged**